invoice_creator.go

```diff
@@ -137,9 +137,12 @@ func (c *InvoiceCreator) Create(amtMSat int64, expiry time.Duration,
 
 	options = append(options, zpay32.Expiry(expiry))
 
-	// Use the memo field as the description. If this is not set
-	// this will just be an empty string.
-	options = append(options, zpay32.Description(memo))
+	// Use the memo field as the description if it's not empty.
+	// Otherwise, we could have the error "both description and description hash set" in zpay32.NewInvoice
+	// even if `memo` is empty.
+	if len(memo) > 0 {
+		options = append(options, zpay32.Description(memo))
+	}
 
 	if descHash != nil {
```

Comment:
Should this line also change? If descHash is provided but memo is also >0, then I think we still end up with both. Instead, I think this line should be:

`if descHash != nil && len(memo) == 0 {`

Reply:
Actually no. We had a conversation with @joostjager and ignoring one of the two field is not a good idea. (even if it is done that way in LND [here](https://github.com/lightningnetwork/lnd/blob/master/lnrpc/invoicesrpc/addinvoice.go#L322).

It should be the responsibility of the application (that uses lnmux) to ensure that it does not give a description and a descriptionhash at the same time

Reply:
Sounds good, I'll merge now.